### DISCUSSION TITLE
Cancellation

### DISCUSSION THREAD

**Discussion by adam12west23** [knaeckeKami/diffutil.dart#6] [Ideas]
Hi,
it would be useful if the computeDiff function could be canceled.
I am trying to perform a diff compute request in an Isolate that executes the algorithm, but if a new request arrives, I would like to cancel the previous one (before it finishes) in order to start the new one as soon as possible.
Thanks

**Comment by knaeckeKami** [comment]
Yeah, I thought about adding better support for running the computations in an isolate, but I didn't find a good API that fits all use cases yet.

Is wrapping the Future in a CancelableOperation good enough for you?

I'm not sure what the best way to support true cancellation would be, as the operation now in synchronous, so there's not really a way to check for cancellation, I would have to make the computation async and then check for cancellation every few steps?

Or maybe just spawn the calculation in an isolate and kill that isolate on cancellation...
